src/AzureAppConfigurationImpl.ts

```diff
@@ -176,34 +176,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         return {
             requestTracingEnabled: this.#requestTracingEnabled,
             initialLoadCompleted: this.#isInitialLoadCompleted,
-            appConfigOptions: this.#options
+            appConfigOptions: this.#options,
+            isFailoverRequest: this.#isFailoverRequest
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute) {
+        const clients = await this.#clientManager.getClients();
+        if (clients.length === 0) {
+            this.#clientManager.refreshClients();
+            throw new Error("No client is available to connect to the target App Configuration store.");
+        }
 
-        // validate selectors
-        const selectors = getValidKeyValueSelectors(this.#options?.selectors);
+        for (const client of clients) {
+            let successful = false;
+            try {
+                const result = await funcToExecute(client.client);

```

Comment:
for (const clientWrapper of clientWrappers)

client.client looks bad. clientWrapper.client is better.